### PR TITLE
geotagging: bug fixes and enhancements

### PR DESCRIPTION
**Pull request by phweyland**
- fix: refresh ui on view change
 - fix: remove redundant signal
 - improve set datetime and geotag (list of images + list of values)
 - fix: avoid useless datetime and geotag undo/redo signals
 - apply geo-location on selected tracks only
 - add button to apply offset and geo-location at once

complete and replace #8243 (with review comments applied) and #8267

### CODE REVIEW

**Comment by phweyland** [comment]
@TurboGit , I don't have any track around Dieppe, but apart this I don't see any other issue. Should be good for your review.

**Review by TurboGit — APPROVED** [review]
Works for me !

File: src/libs/geotagging.c [hunk 1]
```diff
@@ -351,6 +353,72 @@ static void _refresh_images_displayed_on_track(const int segid, const gboolean a
...
+static void _update_buttons(dt_lib_module_t *self)
+{
+  dt_lib_geotagging_t *d = (dt_lib_geotagging_t *)self->data;
+  gtk_widget_set_sensitive(d->map.preview_button, d->map.nb_tracks);
+  GtkWidget *label = gtk_bin_get_child(GTK_BIN(d->map.apply_gpx_button));
+  gtk_label_set_text(GTK_LABEL(label), d->offset ? _("apply offset and geo-location")
+                                                 : _("apply geo-location"));
+  gtk_widget_set_tooltip_text(d->map.apply_gpx_button,
+                              d->offset ? _("apply offset and geo-location to matching images"
+                                            "\ndouble operation: two ctrl-Z to undo")
```
Comment (TurboGit) [inline src/libs/geotagging.c:388]: I'm pretty sure we can do better. Wrapping the changes into a group should do the trick and require a single undo. Can you try this ?

Comment (phweyland) [reply src/libs/geotagging.c:388]: I thought about this but found it a bit tricky.
First a bit of code duplication. In the job for set geolocation we have to add set offset treatment and tweak the messages accordingly.
But then the messages of undo will not accurate anymore (as specific for each operation).
Last a small doubt maybe not founded. If the order for the undos execution is not important, the order for redos is fundamental (set offset before before set geolocation). Should work ?

Still motivated ?

**Pull request merged**